### PR TITLE
ENG-5064 ENG-5065 ENG-5066 - Improvement of APIs for uninstall

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

File: engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentUsageEntity.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public class ComponentUsageEntity implements Serializable {
     public static final String TYPE_CATEGORY = "category";
     public static final String TYPE_WIDGET = "widget";
     public static final String TYPE_FRAGMENT = "fragment";
-    public static final String TYPE_PAGE_TEMPLATE = "pageTemplate";
+    public static final String TYPE_PAGE_TEMPLATE = "pageModel";
```
Comment (entgigi) [inline engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentUsageEntity.java]: **[Major]** this creates confusion, change the constant name and possibly put a comment to describe why you did it (I know there is a ticket but usually when you read code you don't go to see git history and associated ticket)

File: engine/src/main/java/org/entando/entando/aps/system/services/pagemodel/PageModelService.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public class PageModelService implements IPageModelService, ApplicationContextAw
 
     private final EntLogger logger = EntLogFactory.getSanitizedLogger(getClass());
     
-    public static final String TYPE_PAGE_MODEL = "pageModel";
+    public static final String TYPE_PAGE_MODEL = ComponentUsageEntity.TYPE_PAGE_TEMPLATE;
```
Comment (entgigi) [inline engine/src/main/java/org/entando/entando/aps/system/services/pagemodel/PageModelService.java]: **[Minor]** why another constant with the same value? if you really intend to fix things in the future you should use `TYPE_PAGE_TEMPLATE` directly

File: engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java [hunk 3]
```diff
@@ -189,4 +196,30 @@ private boolean checkReferences(List<ComponentDeleteRequestRow> componentsToDele
     static class ComponentDeletionStep {
         private boolean successful;
     }
+    
+    public static class ComponentDeleteRequestRowComparator implements Comparator<ComponentDeleteRequestRow> {
```
Comment (entgigi) [inline engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java]: **[Major]** You should put this inner class inside `ComponentDeleteRequestRow` not in a service, usually we don't put an inner class about DTO in a service, this comparator could be used outside this service.

File: engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java [hunk 4]
```diff
@@ -102,12 +104,17 @@ public ComponentDeleteResponse deleteInternalComponents(List<ComponentDeleteRequ
...
-                            responseRow = service.getComponentDto(code)
-                                    .map(c -> this.extractUsageDetails(code, service))
-                                    .map(usage -> check(components, code, usage))
-                                    .map(c -> delete(c, type, service))
-                                    .orElseGet(() -> ComponentDeleteResponseRow.builder().code(code).type(type)
-                                    .status(ComponentDeleteResponse.STATUS_FAILURE).build());
+                            Optional<IComponentDto> dto = service.getComponentDto(code);
+                            if (dto.isPresent()) {
+                                responseRow = dto.map(c -> this.extractUsageDetails(code, service))
+                                        .map(usage -> check(sortedComponents, code, usage))
```
Comment (entgigi) [inline engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java]: **[Major]** this is a confusing mix of old and new you should rewrite to make it more clear, clean, concise and readable:
```
responseRow = service.getComponentDto(code)
                                    .map(d -> deleteComponentAndBuildResponse(
                                            d, sortedComponents, code, type, service
                                    ))
                                    .orElseGet(() -> ComponentDeleteResponseRow.builder().code(code).type(type)
                                            .status(ComponentDeleteResponse.STATUS_SUCCESS).build());
-----
private ComponentDeleteResponseRow deleteComponentAndBuildResponse(IComponentDto dto,
                                                                       List<ComponentDeleteRequestRow> sortedComponents,
                                                                       String code,
                                                                       String type, IComponentUsageService service) {
        return Optional.of(dto).map(c -> this.extractUsageDetails(code, service))
                .map(usage -> check(sortedComponents, code, usage))
                .map(c -> delete(c, type, service))
                .orElseGet(() -> ComponentDeleteResponseRow.builder().code(code).type(type)
                        .status(ComponentDeleteResponse.STATUS_FAILURE).build());
    }
```

**Review by entgigi — CHANGES_REQUESTED** [review]

**Review by mirkoPan — APPROVED** [review]
LGTM

File: engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java (hunk 4 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java]: Done, good advice

File: engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java (hunk 3 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java]: Done

File: engine/src/main/java/org/entando/entando/aps/system/services/pagemodel/PageModelService.java (hunk 2 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/pagemodel/PageModelService.java]: Done

File: engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentUsageEntity.java (hunk 1 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentUsageEntity.java]: Done

**Review by entgigi — APPROVED** [review]
LGTM

**Pull request merged**